### PR TITLE
Add CallTarget support to MongoDB

### PR DESCRIPTION
**Pull request by kevingosse**

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/MongoDb/MongoDbInstrumentMethodAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.MongoDb
+{
+    internal abstract class MongoDbInstrumentMethodAttribute : InstrumentMethodAttribute
+    {
+        protected MongoDbInstrumentMethodAttribute(string typeName)
+        {
+            AssemblyName = MongoDbIntegration.MongoDbClientAssembly;
+            TypeName = typeName;
+            IntegrationName = MongoDbIntegration.IntegrationName;
+            ParameterTypeNames = new[] { "MongoDB.Driver.Core.Connections.IConnection", ClrNames.CancellationToken };
```
Comment (zacharycmontoya) [inline src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/MongoDb/MongoDbInstrumentMethodAttribute.cs:12]: nit: Would you mind moving the `ParameterTypeNames` assignment into each of the method-specific attributes? That could reduce confusion if we later decided to instrument other methods in MongoDb

File: src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/MongoDb/IWireProtocol_Generic_Execute_Integration.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        typeName: "MongoDB.Driver.Core.WireProtocol.GetMoreWireProtocol`1",
+        isGeneric: true)]
+    [MongoDbExecute(
+        typeName: "MongoDB.Driver.Core.WireProtocol.QueryWireProtocol`1",
+        isGeneric: true)]
+    [MongoDbExecute(
+        typeName: "MongoDB.Driver.Core.WireProtocol.WriteWireProtocolBase`1",
+        isGeneric: true)]
+    // ReSharper disable once InconsistentNaming
+    public class IWireProtocol_Generic_Execute_Integration
```
Comment (zacharycmontoya) [inline src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/MongoDb/IWireProtocol_Generic_Execute_Integration.cs:30]: I love the simplicity of this file 😍

File: src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/MongoDb/MongoDbInstrumentMethodAttribute.cs (hunk 1 above)
Comment (kevingosse) [reply src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/MongoDb/MongoDbInstrumentMethodAttribute.cs]: Done in https://github.com/DataDog/dd-trace-dotnet/pull/1214/commits/ff06266d093c73f6d518c8aca8c7d2017e203936

**Review by tonyredondo — APPROVED** [review]
LGTM

**Pull request merged**